tutorials/get-started/index.md

```diff
@@ -0,0 +1,989 @@
+        title: new Text('Startup Name Generator'),
+      ),
+      body: _buildSuggestions(),
+    );
+  }
+
+  Widget _buildSuggestions() {
+    return new ListView.builder(
+        padding: const EdgeInsets.all(16.0),
+        itemBuilder: (context, i) {
+          if (i.isOdd) return new Divider();
+
+          final index = i ~/ 2;
+          if (index >= _suggestions.length) {
+            _suggestions.addAll(generateWordPairs().take(10));
+          }
+          return _buildRow(_suggestions[index]);
+        });
+  }
+
+  Widget _buildRow(WordPair pair) {
+    return new ListTile(
+        title: new Text(
+      pair.asPascalCase,
+      style: _biggerFont,
+    ));
+  }
+}
+{% endprettify %}
+
+---
+
+## Step 5: Save favorite word pairings
+
+In this step, you'll add tappable heart icons to each row.
+When the user taps an entry in the list, toggling its
+"favorited" state, that word pairing is added or removed from a
+set of saved favorites.
+
+<!--
+- Add ability to save favorites (in state class)
+- Add navigation to view saved word pairings
+-->
+
+<ol markdown="1">
+<li markdown="1"> Add a `_saved` Set to RandomWordsState. This Set stores
+    the word pairings that the user favorited. A Set is used instead of a List
+    because a Set typically doesn't allow duplicates, if it's implemented not to
+    do so.
+
+<!-- skip -->
+{% prettify dart %}
+class RandomWordsState extends State<RandomWords> {
+  final _suggestions = <WordPair>[];
+
+  [[highlight]]final _saved = new Set<WordPair>();[[/highlight]]
+
+  final TextStyle _biggerFont = new TextStyle(fontSize: 18.0);
+  ...
+}
+{% endprettify %}
+</li>
+
+<li markdown="1">
+In the `_buildRow` function, add `alreadySaved`, a check to ensure that
+a word pairing hasn't already been added to favorites.
+
+<!-- skip -->
+{% prettify dart %}
+  Widget _buildRow(WordPair pair) {
+    [[highlight]]final alreadySaved = _saved.contains(pair);[[/highlight]]
+    ...
+  }
+{% endprettify %}
+</li>
+
+<li markdown="1">
+Also in `_buildRow`, add heart-shaped icons to the ListTiles to enable
+favoriting. Display favorite entries with a solid red heart,
+and other entries with a heart-shaped outline.
+
+Make the highlighted changes below:
+
+<!-- skip -->
+{% prettify dart %}
+  Widget _buildRow(WordPair pair) {
+    final alreadySaved = _saved.contains(pair);
+    return new ListTile(
+        title: new Text(
+          pair.asPascalCase,
+          style: _biggerFont,
+        [[highlight]]),[[/highlight]]  // Replace the "));" with this.
+        [[highlight]]trailing: new Icon([[/highlight]] // Add this line, plus the next 3 lines.
+          [[highlight]]alreadySaved ? Icons.favorite : Icons.favorite_border,[[/highlight]]
+          [[highlight]]color: alreadySaved ? Colors.red : null,[[/highlight]]
+        [[highlight]]));[[/highlight]]
+  }
+}
+{% endprettify %}
+</li>
+
+<li markdown="1">
+Run the app. You should now see open hearts on each row, but they aren't yet
```

Comment:
Maybe we should clarify when the user needs to "run/restart" the app vs "hot reload" their app.

Reply:
Yup. See comment below.